INF-48: Shift Displayr builds to Github Actions

mitchell-demol-displayr opened the pull request:
Changed the chrometests to compress with tgz and separated a large dll from it to speed up the process.
Have added code to allow both appveyor and github actions to run the tests accounting for theses changes.
Also changed the rerun download code to just be a single step.

mitchell-demol-displayr committed:
Update test_location
must update as we have changed the path of the tests folder in the zip

mitchell-demol-displayr committed:
Folder is just 'outputs'

mitchell-demol-displayr committed:
debug test failure

mitchell-demol-displayr committed:
fix PS syntax

mitchell-demol-displayr committed:
Try create chrometests/bin/release
Maybe tests expect us to be in this dir and uses relative paths

mitchell-demol-displayr committed:
try x86 option to run 32bit

mitchell-demol-displayr committed:
Can't run x86

mitchell-demol-displayr committed:
split sharedwebui from chrometest folder

mitchell-demol-displayr committed:
dont cd to repo, need to extract to right path

mitchell-demol-displayr committed:
extract files into test folder

mitchell-demol-displayr committed:
change test timeout back to 30mins

mitchell-demol-displayr committed:
Not zipping dll, must change here

mitchell-demol-displayr committed:
update to tgz

mitchell-demol-displayr committed:
Re-enable reruns of chrometests

mitchell-demol-displayr committed:
run only specific tests

mitchell-demol-displayr committed:
fix nunit cmd line syntax

mitchell-demol-displayr committed:
disable batches and dont report to database

mitchell-demol-displayr committed:
fix syntax

refer to inputs with just inputs.x

mitchell-demol-displayr committed:
wrong cli option

mitchell-demol-displayr committed:
fix cli syntax

mitchell-demol-displayr committed:
Let artifacts be uploaded

mitchell-demol-displayr committed:
remove result xml causing error

mitchell-demol-displayr committed:
revert back to run all batches

mitchell-demol-displayr committed:
Accommodate appveyor and GHA chrometest downloads

Also incorporated retry of downloads into step - refactored code from master branch

mitchell-demol-displayr committed:
Remove redundant code

mitchell-demol-displayr committed:
remove verbose output

mitchell-demol-displayr committed:
let database report step run

mitchell-demol-displayr commented on .github/workflows/chrometest_run-batches.yml:
```diff
@@ -114,9 +119,28 @@ jobs:
-          $uri = "${{ env.ENDPOINT_CHROMETEST }}/${{ env.BUILD_NUMBER }}/ChromeTests.7z?${{ secrets.ACCESS_TOKEN_CONTAINER_BUILDS }}"
-          Invoke-WebRequest -Uri $uri -OutFile ChromeTests.7z
-          7z x -y .\ChromeTests.7z
+          # retry in case of failed download
```
Rerun change was added after I made this branch, I just added this into a single step here

mlau-displayr replied:
Nice

mitchell-demol-displayr commented on .github/workflows/chrometest_run-batches.yml:
```diff
@@ -142,6 +166,27 @@ jobs:
           .\nunit3-console.exe ChromeTests.dll --encoding=utf-8 --domain=None --inprocess --labels=All --testparam non-interactive=true `
             --result=${{ env.RERUN_RESULT_FILE }} `
             --testlist $OriginalFailTestResult
+      # Assuming average of 30 chrometests per day, total size of text files for a run should be ~300 kb, these will expire after 7 days, so 7x30x300 = 63 MB used at any time
```
Can remove this, basically just repeating the rerun step on job reruns. Need to cache the result file to do so.

mlau-displayr replied:
We can do this a bit later.

Lets just make sure the chrometest are stable first. Cause we are making a big change atm

mitchell-demol-displayr committed:
remove rerun code

mitchell-demol-displayr committed:
removed more rerun code

mitchell-demol-displayr committed:
revert this small change

mitchell-demol-displayr committed:
Updating refs to source to new name

Changing the input to take either "Appveyor" or  "Github Actions".
Leaving code to allow for "gha" and "appveyor" to reduce issues.

mlau-displayr committed:
Merge branch 'master' of github.com:Displayr/gha-templates into INF-48

mlau-displayr committed:
FIX update rerun folder deletion

mitchell-demol-displayr committed:
Remove conflicting code from merge

I already encapsulated retry downloads, 
remove the old code that got re-added in merge. It causes that step to fail for GHA dispatches.
Added some logging.

mitchell-demol-displayr committed:
Fixed chrometest download retry logic

Would throw error on first failure previously, now it should successfully loop and throw error only if all attempts fail

mitchell-demol-displayr committed:
Fix 7z errors aren't throwing errors

Manually check for 7z error and throw one ourselves so loop actually works

mitchell-demol-displayr committed:
Fix removal of artifacts folder

mitchell-demol-displayr committed:
Set timeout on download + revert to 2 steps

Jobs are stalling on this step
reverting to two steps 
also removing redundant code which facilitates github actions

mitchell-demol-displayr committed:
Allow CircleCI builds to run chrometests

CircleCI chrometests are compressed into tgz like GHA

mitchell-demol-displayr committed:
Debug dl problem

+ fix throw syntax

mitchell-demol-displayr committed:
fix errors

was in dir we were checking for using relative path, so error, just dl file into folder instead of cding there

mitchell-demol-displayr committed:
Add timeout to rerun download 

Had a chrometest stuck on this step for 3hours
